utilities/transactions/pessimistic_transaction.cc

```diff
@@ -555,6 +559,10 @@ Status PessimisticTransaction::ValidateSnapshot(
   assert(snapshot_);
 
   SequenceNumber seq = snapshot_->GetSequenceNumber();
+  // TODO(myabandeh): Since the first time the new_seqno is set to the snapshot
+  // sequence number, and snapshot sequence number can only increase by new
+  // snapshots, I am not sure how this inequality could be false except for the
+  // inital case where pref_seqno is kMaxSequenceNumber.
   if (prev_seqno <= seq) {
```

Comment:
@igorcanadi If my understanding is correct, I think it would simplify the logic if we explicitly check against kMaxSequenceNumber rather than a general inequality check.

Reply:
This sounds correct. I would change the `if` statement to `if (prev_seqno != kMaxSequenceNumber)` and then make an assertion `assert(prev_seqno <= seq)` just to catch any future mistakes.